lib/Sema/CSDiagnostics.cpp

```diff
@@ -3168,6 +3168,14 @@ bool ContextualFailure::diagnoseConversionToBool() const {
     return true;
   }
 
+  if (auto fnType = getFromType()->getAs<FunctionType>()) {
+    if (fnType->getResult()->isBool() && fnType->getNumParams() == 0) {
+      emitDiagnostic(diag::missing_nullary_call, toType)
+          .fixItInsertAfter(getSourceRange().End, "()");
+      return true;
+    }
+  }
+
```

Comment:
Ideally this would be handled by `MissingCallFailure`, to record the fix for that I believe you should to be able to add:

```cxx
    if (repairByInsertingExplicitCall(lhs, rhs))
      return true;
```

in `repairFailures` for the `case ConstraintLocator::Condition` case

Reply:
Thanks for the suggestion, did it this way.